lib/transaction/trace/index.js

```diff
@@ -83,6 +83,16 @@ Trace.prototype.generateSpanEvents = function generateSpanEvents() {
 
     const children = this.root.getChildren()
     for (let i = 0; i < children.length; ++i) {
+      if (i === 0) {
```

Comment:
This is kind of ugly.  Should this be moved to its own function?

Reply:
no reason to have in the loop, would pull out prior to the loop.

```
firstChild = children[0]
if (firstChild) {
  // apply trace attributes
```

beyond that, a helper method would probably be nice for what is there.

i'm wondering if the attribute might get filtered before where you are grabbing it? for example, if the transaction filtering has it off but it isn't filtered out of spans, i don't think it will retrieve the item when it should. 

it really is awkward where trace is conjuring this but would prob take a lot of digging to find a better spot, unless someone else knows off hand a better location. as such, you might consider caching the value on the trace at the point it adds the original attribute and then checking for it existing or not.

Reply:
Is this to get the root span?

Reply:
It's to grab the first span to be processed.

Reply:
also, commenting why this is happening would be good i.e. at the point it is determined there is only a root span so this adds to the first non-root.

Reply:
I think I will try caching the value where its added.  That seems much cleaner.